test/examples/ucp_client_server.c

```diff
@@ -220,23 +225,24 @@ static int send_recv_stream(ucp_worker_h ucp_worker, ucp_ep_h ep, int is_server)
                     ucs_status_string(UCS_PTR_STATUS(request)));
             ret = -1;
             goto out;
-        } else if (UCS_PTR_STATUS(request) != UCS_OK) {
+        } else {
             request_wait(ucp_worker, request);
         }
     } else {
         /* Server receives a message from the client using the stream API */
         request = ucp_stream_recv_nb(ep, &recv_message, 1,
                                      ucp_dt_make_contig(TEST_STRING_LEN),
-                                     stream_recv_cb, &length , 0);
+                                     stream_recv_cb, &length,
+                                     UCP_STREAM_RECV_FLAG_WAITALL);
         if (UCS_PTR_IS_ERR(request)) {
             fprintf(stderr, "unable to receive UCX message (%s)\n",
                     ucs_status_string(UCS_PTR_STATUS(request)));
             ret = -1;
             goto out;
         } else {
             request_wait(ucp_worker, request);
-            printf("UCX data message was received\n");
         }
+        printf("UCX data message was received\n");
```

Comment:
why did you move the `printf`?
in general, request can be completed with error, lets return actual status from `request_wait` and print the message only if it's `UCS_OK`. you can obtain the status by `ucp_request_check_status` before freeing.

Reply:
After I added check for UCS_OK send and recv become pretty similar, so I moved status check code into wait function. Is it better now?